sympy/tensor/indexed.py

```diff
@@ -437,10 +437,9 @@ def __new__(cls, label, shape=None, **kw_args):
 
         assumptions, kw_args = _filter_assumptions(kw_args)
         if isinstance(label, string_types):
-            label = Symbol(label)
+            label = Symbol(label, **assumptions)
         elif isinstance(label, Symbol):
             assumptions = label._merge(assumptions)
-            label = Symbol(label.name)
```

Comment:
What about if you specify additional assumptions in `IndexedBase` that are not in the label? Is `_merge` also adding the additional assumptions to `label`?